### ISSUE TITLE
request for a more detailed description of BERTNLU

### ISSUE THREAD

**Issue by IreneSucameli** [thu-coai/ConvLab-2#218]
Hi, it would be possible to have more explaination about how BERTNLU works?
Specifically, it is not clear if:

- it is a multi-task model or not. Is it possible to have a more detailed description of the model?
- what the "fine-tune" , the "context" and the "context_grad" hyperparameters do in BERTNLU. It seems to me that "fine-tune: false" allows to freeze the LM weights for slots training, while "context_grad: false" do the same for intents. Is that correct? What does "context" do, then?

Thanks

**Comment by zqwerty** [comment]
For model details, [code](https://github.com/thu-coai/ConvLab-2/blob/master/convlab2/nlu/jointBERT/jointBERT.py) explains well.

It is a multi-task model.
- `fine-tune`: whether tunes the bert parameter. if false, we only tune added classification layers.
- `context`: whether use context information. if false, we pass [CLS] representation of the single utterance to the intent classifier and pass tokens' representations to the slot classifier. if true, we would use the same bert to encode concatenated context utterances and obtain its [CLS] representation as **context representation**, which would be concated to **every** [CLS] and tokens' representations of current utterance for intent/slot classification.
- `context_grad`: whether calculate the grad through **context representation**. if false, the loss will not back-propagate to context encoder.
